From 904002c1482f18dcefcdfddab6834d44cb62cb84 Mon Sep 17 00:00:00 2001
From: Christian Ambach <ambi@samba.org>
Date: Wed, 11 May 2016 17:41:24 +0200
Subject: [PATCH 1090/1096] s3:rpcclient make --pw-nt-hash option work

Bug: https://bugzilla.samba.org/show_bug.cgi?id=10796

Signed-off-by: Christian Ambach <ambi@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>
(cherry picked from commit 4fe59879cc2a608194578e33e27e0dc1e2f0fc58)
(cherry picked from commit 2f579c7b40738e23691f28eb1611427708ef8258)
---
 source3/rpcclient/rpcclient.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/source3/rpcclient/rpcclient.c b/source3/rpcclient/rpcclient.c
index a4edbe8..74ae708 100644
--- a/source3/rpcclient/rpcclient.c
+++ b/source3/rpcclient/rpcclient.c
@@ -1095,6 +1095,9 @@ out_free:
 	if (get_cmdline_auth_info_use_ccache(rpcclient_auth_info)) {
 		flags |= CLI_FULL_CONNECTION_USE_CCACHE;
 	}
+	if (get_cmdline_auth_info_use_pw_nt_hash(rpcclient_auth_info)) {
+		flags |= CLI_FULL_CONNECTION_USE_NT_HASH;
+	}
 
 	user = talloc_strdup(frame, get_cmdline_auth_info_username(rpcclient_auth_info));
 	SMB_ASSERT(user != NULL);
-- 
2.9.0

